compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/tree/nodeparser/ParseIntermediateClauseTest.java

```diff
@@ -0,0 +1,876 @@
+    }
+
+    @Test
+    public void testValidOrderClause4() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by name " +
+                "ascending, emp.id ascending");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause5() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by name" +
+                " ascending, emp.id descending");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit 1");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause2() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit getName()");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause3() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit emp.id");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    // Invalid syntax tests
+
+    @Test
+    public void testEmptyString() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.WHERE_CLAUSE);
+        Assert.assertTrue(intermediateNode.hasDiagnostics());
+        Assert.assertEquals(intermediateNode.leadingInvalidTokens().size(), 0);
+        Assert.assertEquals(intermediateNode.trailingInvalidTokens().size(), 0);
+        Assert.assertEquals(intermediateNode.toString(), " MISSING[where] MISSING[]");
+    }
+
+    @Test
+    public void testEmptyFromClause() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("from");
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.FROM_CLAUSE);
+        Assert.assertTrue(intermediateNode.hasDiagnostics());
+        Assert.assertEquals(intermediateNode.leadingInvalidTokens().size(), 0);
+        Assert.assertEquals(intermediateNode.trailingInvalidTokens().size(), 0);
+        Assert.assertEquals(intermediateNode.toString(), "from MISSING[] MISSING[] MISSING[in] MISSING[]");
+    }
+
+    @Test
+    public void testFromClauseWithInvalidKeyword2() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("from int");
```

Comment:
test case does not have an invalid keyword right? although the test name suggests otherwise.
InvalidKeyword/IncorrectKeyword is when the recovered tree has invalid tokens. This test only has missing tokens. 

Also, let's only use one term. I prefer "InvalidTokens" which includes things like invalid `%` etc.

Reply:
Updated